packages/twenty-front/src/modules/views/hooks/internal/useFiltersFromQueryParams.ts

```diff
@@ -0,0 +1,142 @@
+import { ViewFilter } from '@/views/types/ViewFilter';
+import { ViewFilterOperand } from '@/views/types/ViewFilterOperand';
+import { assertNotNull } from '~/utils/assert';
+
+const filterQueryParamsSchema = z.object({
+  filter: z.record(
+    z.record(
+      z.nativeEnum(ViewFilterOperand),
+      z.string().or(z.array(z.string())),
+    ),
+  ),
+});
+
+export type FilterQueryParams = z.infer<typeof filterQueryParamsSchema>;
+
+export const useFiltersFromQueryParams = () => {
+  const apolloClient = useApolloClient();
+  const [searchParams] = useSearchParams();
+  const { objectNamePlural = '' } = useParams();
+  const { objectNameSingular } = useObjectNameSingularFromPlural({
+    objectNamePlural,
+  });
+  const { objectMetadataItem } = useObjectMetadataItem({ objectNameSingular });
+  const generateFindManyRecordsQuery = useGenerateFindManyRecordsQuery();
+
+  const filterParamsValidation = filterQueryParamsSchema.safeParse(
+    qs.parse(searchParams.toString()),
+  );
+  const filterQueryParams = useMemo(
+    () =>
+      filterParamsValidation.success ? filterParamsValidation.data.filter : {},
+    [filterParamsValidation],
+  );
+  const hasFiltersQueryParams = filterParamsValidation.success;
+
+  const getFiltersFromQueryParams = useRecoilCallback(
+    ({ snapshot }) =>
+      async () => {
+        if (!hasFiltersQueryParams) return [];
+
+        return (
+          await Promise.all(
+            Object.entries(filterQueryParams).map<Promise<ViewFilter | null>>(
+              async ([fieldName, paramValue]) => {
+                const [operand, value] = Object.entries(paramValue)[0];
+                const fieldMetadataItem = objectMetadataItem.fields.find(
+                  (field) => field.name === fieldName,
+                );
+
+                if (!fieldMetadataItem) return null;
+
+                const filterDefinition =
+                  formatFieldMetadataItemAsFilterDefinition({
+                    field: fieldMetadataItem,
+                  });
+
+                if (!filterDefinition) return null;
+
+                const relationObjectMetadataNameSingular =
+                  fieldMetadataItem.toRelationMetadata?.fromObjectMetadata
+                    .nameSingular;
+                const relationObjectMetadataNamePlural =
+                  fieldMetadataItem.toRelationMetadata?.fromObjectMetadata
+                    .namePlural;
+                const relationObjectMetadataItem =
+                  relationObjectMetadataNameSingular
+                    ? snapshot
+                        .getLoadable(
+                          objectMetadataItemFamilySelector({
+                            objectName: relationObjectMetadataNameSingular,
+                            objectNameType: 'singular',
+                          }),
+                        )
+                        .getValue()
+                    : null;
+                const relationRecordNames =
```

Comment:
This oneliner is difficult to read, could we "unwrap" it on multiple lines with more self-explanatory block statements ? Maybe more if block statements and even using a let variable would make this part easier to read, multiple simple statements are easier to read than a concise one liner.